src/backend/cdb/cdbpathlocus.c

```diff
@@ -432,26 +432,17 @@ cdbpathlocus_from_subquery(struct PlannerInfo *root,
 		 */
 		if (rel->reloptkind == RELOPT_OTHER_MEMBER_REL)
 		{
-			Index		parent_relid = 0;
+			Index parent_relid = 0;
```

Comment:
How about code like this?

```suggestion
			Index parent_relid;
			if (root->append_rel_array == NULL || root->append_rel_array[rel->relid] == NULL)
			    elog(ERROR, "cannot find valid append rel");
             parent_relid = root->append_rel_array[rel->relid];
```

Also remove the if-clause `if (parent_relid <= 0)`.

Reply:
Good suggestion, edited.